src/cmd/serial.js

```diff
@@ -233,133 +246,265 @@ module.exports = class SerialCommand extends CLICommandBase {
+				this.ui.stdout.write(`Interface is ${_.capitalize(currIfaceName)}${os.EOL}`);
+			} else {
+				this.ui.stdout.write(`Your device MAC address is ${chalk.bold.cyan('00:00:00:00:00:00')}${os.EOL}`);
+			}
+		} catch (err) {
+			throw new VError(ensureError(err), 'Could not get MAC address');
+		} finally {
+			if (device && device.isOpen) {
+				await device.close();
+			}
+		}
 	}
 
-	inspectDevice({ port }){
-		return this.whatSerialPortDidYouMean(port, true)
-			.then(device => {
-				if (!device){
-					throw new VError('No serial port identified');
-				}
+	/**
+	 * Inspects a Particle device and provides module info and asset info
+	 * @param {string} port
+	 */
+	async inspectDevice({ port }) {
+		let deviceFromSerialPort, deviceId, device;
+
+		try {
+			deviceFromSerialPort = await this.whatSerialPortDidYouMean(port, true);
+			deviceId = deviceFromSerialPort.deviceId;
+			device = await usbUtils.getOneUsbDevice({ idOrName: deviceId });
+		} catch (err) {
+			throw new VError(ensureError(err), 'Could not get inspect device');
+		}
 
-				return this.getSystemInformation(device);
-			})
-			.then((data) => {
-				const functionMap = {
-					s: 'System',
-					u: 'User',
-					b: 'Bootloader',
-					r: 'Reserved',
-					m: 'Monolithic',
-					a: 'Radio stack',
-					c: 'NCP'
-				};
-				const locationMap = {
-					m: 'main',
-					b: 'backup',
-					f: 'factory',
-					t: 'temp'
-				};
 
-				const d = JSON.parse(data);
-				const parser = new DescribeParser();
-				const modules = parser.getModules(d);
+		let platformName = null;
+		let platformId = null;
+		try {
+			platformName = deviceFromSerialPort.specs.productName;
+			platformId = deviceFromSerialPort.specs.productId;
+		} catch (err) {
+			// ignore error and move on to get other fields
+		}
+		this.ui.stdout.write(`Platform : ${platformId} - ${chalk.bold.cyan(platformName)}${os.EOL}${os.EOL}`);
+
+		try {
+			this._getModuleInfo(device);
+		} catch (err) {
+			throw new VError(ensureError(err), 'Could not get inspect device');
+		} finally {
+			if (device && device.isOpen) {
+				await device.close();
+			}
+		}
+	}
+	
+	/**
+	 * Obtains module info from control requests
+	 * @param {object} device
+	 * @returns {boolean} returns error or true
+	 */
+	async _getModuleInfo(device) {
+		// TODO: Get this from the protobuf exports
+		const fwModuleDisplayNames = {
+			'INVALID': 'Invalid',
+			'RESOURCE': 'Resource',
+			'BOOTLOADER': 'Bootloader',
+			'MONO_FIRMWARE': 'Monolithic Firmware',
+			'SYSTEM_PART': 'System Part',
+			'USER_PART': 'User Part',
+			'SETTINGS': 'Settings',
+			'NCP_FIRMWARE': 'Network Co-processor Firmware',
+			'RADIO_STACK': 'Radio Stack Module',
+			'ASSET': 'Asset'
+		};
 
-				if (d.p !== undefined){
-					const platformName = knownPlatformDisplayForId()[d.p];
-					console.log('Platform:', d.p, platformName ? ('- ' + chalk.bold.cyan(platformName)) : '');
+		const modules = await device.getFirmwareModuleInfo();
```

Comment:
This should be a TODO in particle-usb 😛 => This method doesn't have timeout, sometimes if the device is "busy" this call will last for long time hehe. Is not intentional that doesn't have timeout, when I created that function I didn't know how the things worked in particle-usb. So in summary: We need to add timeout as param to this function 😄